### PR TITLE
Fix #4: Configure persistence

### LINKED ISSUE #4: Configure persistence
Introduce JPA entities and liquibase configuration.
Endpoints from #3 will work with persisted data.

### PR DESCRIPTION
**Pull request by banterCZ**

### CODE REVIEW

File: doc/sql/postgresql/create-schema.sql [hunk 1]
```diff
@@ -0,0 +1,25 @@
+CREATE TABLE public.ud_user_claims
```
Comment (romanstrobl) [inline doc/sql/postgresql/create-schema.sql:1]: I think we should not define the schema here (`public`).

File: doc/sql/postgresql/create-schema.sql [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+(
+    user_id           VARCHAR(255) NOT NULL,
+    claims            TEXT         NOT NULL,
+    timestamp_created TIMESTAMP WITHOUT TIME ZONE DEFAULT NOW(),
+    timestamp_updated TIMESTAMP WITHOUT TIME ZONE,
+    CONSTRAINT ud_user_claims_pkey PRIMARY KEY (user_id)
+);
+
+-- Spring Security
+CREATE TABLE ud_users
```
Comment (romanstrobl) [inline doc/sql/postgresql/create-schema.sql:11]: Can we use `uds` as acronym?

File: doc/sql/postgresql/create-schema.sql [hunk 3]
```diff
@@ -0,0 +1,25 @@
+CREATE TABLE public.ud_user_claims
+(
+    user_id           VARCHAR(255) NOT NULL,
+    claims            TEXT         NOT NULL,
+    timestamp_created TIMESTAMP WITHOUT TIME ZONE DEFAULT NOW(),
+    timestamp_updated TIMESTAMP WITHOUT TIME ZONE,
```
Comment (romanstrobl) [inline doc/sql/postgresql/create-schema.sql:6]: We usually use `timestamp_last_updated`.

File: docs-private/Developer-How-To-Start.md [hunk 4]
```diff
@@ -6,3 +6,31 @@
...
+
+
+## Database
+
+Database changes are driven by Liquibase.
+
+This is an example how to manually check the Liquibase status.
+Important and fixed parameter is `changeloge-file`.
+Others (like URL, username, password) depend on your environment.
+Mind that the working directory MUST be `src/main/resources` to be consistent with Spring Boot naming (file names are part of the checksum).
```
Comment (romanstrobl) [inline docs-private/Developer-How-To-Start.md:18]: I think no need to capitalize `MUST`.

File: src/main/java/com/wultra/security/userdatastore/userclaims/UserClaimsEntity.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    private String claims;
+
+    private LocalDateTime timestampCreated = LocalDateTime.now();
+
+    private LocalDateTime timestampUpdated;
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
```
Comment (romanstrobl) [inline src/main/java/com/wultra/security/userdatastore/userclaims/UserClaimsEntity.java:52]: I think there are better ways to handle the class comparison. This comparison does not e.g. take into account proxies.

File: src/main/java/com/wultra/security/userdatastore/userclaims/UserClaimsEntity.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+class UserClaimsEntity {
+
+    @Id
+    private String userId;
+
+    private String claims;
+
+    private LocalDateTime timestampCreated = LocalDateTime.now();
+
+    private LocalDateTime timestampUpdated;
```
Comment (romanstrobl) [inline src/main/java/com/wultra/security/userdatastore/userclaims/UserClaimsEntity.java:47]: Let's use `timestampLastUpdated`.

File: src/main/resources/application.properties [hunk 7]
```diff
@@ -2,7 +2,12 @@ spring.application.name=user-data-store
 
 spring.jpa.hibernate.ddl-auto=none
 spring.jpa.open-in-view=false
+spring.jpa.properties.hibernate.connection.provider_disables_autocommit=true
```
Comment (romanstrobl) [inline src/main/resources/application.properties:5]: I have one concern: shouldn't this setting be applied across all backends? Not sure we want to do it differently in each project.

**Review by romanstrobl — CHANGES_REQUESTED** [review]
I found several possible improvements to consider.

File: src/main/resources/application.properties (hunk 7 above)
Comment (banterCZ) [reply src/main/resources/application.properties:5]: I would recommend that. See https://vladmihalcea.com/why-you-should-always-use-hibernate-connection-provider_disables_autocommit-for-resource-local-jpa-transactions/

File: doc/sql/postgresql/create-schema.sql (hunk 2 above)
Comment (banterCZ) [reply doc/sql/postgresql/create-schema.sql:11]: Why not? I was not sure about this first idea.

File: docs-private/Developer-How-To-Start.md (hunk 4 above)
Comment (banterCZ) [reply docs-private/Developer-How-To-Start.md:18]: Ok, too strict. I was thinking about RFC 2119

File: src/main/java/com/wultra/security/userdatastore/userclaims/UserClaimsEntity.java (hunk 5 above)
Comment (banterCZ) [reply src/main/java/com/wultra/security/userdatastore/userclaims/UserClaimsEntity.java:52]: I will regenerate it with the option _accept subclasses_.

File: src/main/resources/application.properties (hunk 7 above)
Comment (romanstrobl) [reply src/main/resources/application.properties:5]: Either we should remove the setting here or file issues to change it in all backends. I don't like the state when it is fixed in one application and stays original in others.

**Review by romanstrobl — APPROVED** [review]
I have just one comment related to the auto-commit behaviour.

File: src/main/resources/application.properties (hunk 7 above)
Comment (banterCZ) [reply src/main/resources/application.properties:5]: Ok, agreed, let's test it here. After that revert or introduce it to all components. I have opened an internal issue https://github.com/orgs/wultra/projects/29/views/1?pane=issue&itemId=21830474

**Pull request merged**